allensdk/test/brain_observatory/filtered_running_speed/test_filtered_running_speed_module.py

```diff
@@ -0,0 +1,145 @@
+import os
+from pathlib import Path
+import json
+import subprocess as sp
+
+import pytest
+import pandas as pd
+
+
+@pytest.fixture
+def use_temp_dir(tmpdir_factory):
+    def fn(
+        data_dir,
+        tempdir_name,
+        input_json_fname,
+        output_json_fname,
+        module,
+        renamer_cb
+    ):
+        """
+        Test the filtered running speed filtered_running_speed
+        module by creating inputs in a temp directory and
+        then by running the module using command line arguments
+        """
+
+        temp_dir = str(tmpdir_factory.mktemp(tempdir_name))
+
+        input_json_path = os.path.join(data_dir, input_json_fname)
+        new_input_json_path = os.path.join(temp_dir, input_json_fname)
+
+        output_json_path = os.path.join(temp_dir, output_json_fname)
+
+        with open(input_json_path, 'r') as input_json:
+            input_json_data = json.load(input_json)
+
+        input_json_data = renamer_cb(input_json_data, data_dir, temp_dir)
+
+        with open(new_input_json_path, 'w') as new_input_json:
+            json.dump(input_json_data, new_input_json)
+
+        process = sp.Popen([
+            'python', '-m', module,
+            '--input_json', new_input_json_path,
+            '--output_json', output_json_path,
+        ], stdout=sp.PIPE, stderr=sp.PIPE)
+
+        process.wait()
+
+        (stdout, stderr) = process.communicate()
+
+        if process.returncode != 0:
+            raise Exception(stderr)
+
+        with open(output_json_path, 'r') as output_json:
+            output_json_data = json.load(output_json)
+
+        return output_json_data
+
+    return fn
+
+
+DATA_DIR = os.environ.get(
+    "ECEPHYS_PIPELINE_DATA",
+    os.path.join(
+        "/",
+        "allen",
+        "aibs",
+        "informatics",
+        "module_test_data",
+        "ecephys",
+        "filtered_running_speed"
+    ),
+)
+
+
+def reparent(path, new_parent):
+    return str(Path(new_parent) / Path(path).name)
+
+
+@pytest.mark.requires_bamboo
+@pytest.mark.parametrize('input_json_fname,output_json_fname', [
+    [
+        "FILTERED_RUNNING_SPEED_QUEUE_input.json",
+        'FILTERED_RUNNING_SPEED_QUEUE_output.json'
+    ]
+])
+def test_extract_running_speed_module(
```

Comment:
This looks like just a smoke test (a test of "does the CLI module run end to end without crashing")

I would feel better if the methods in `filtered_running_speed.py` could be unit tested to make sure that the output for specified input is as expected.

It does not look like many of them *need* to be member methods of the `FilteredRunningSpeed` class (e.g. `get_frame_times` just needs a path to a sync file as an argument). If they were split out as functions, they could be tested individually to make sure that they are performing as expected.

Reply:
Noted. I'll add more tests and remove the Popen call.

Reply:
I have removed the Popen command and I have updated the test as you suggested. You're right - this is a much better way to do it. Thanks. I've also added a few extra tests to test individual methods.